src/app/shared/editor/components/yaml-editor/yaml-editor.component.ts

```diff
@@ -0,0 +1,54 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, OnChanges, Output, SimpleChanges } from "@angular/core";
+
+import * as monaco from "monaco-editor";
+import { MonacoService } from "../../services/monaco.service";
+import { BaseEditorComponent } from "../base-editor/base-editor.componet";
+import { getError } from "../../helpers/editor-error-formatter";
+
+const YAML_EDITOR_OPTIONS: monaco.editor.IStandaloneEditorConstructionOptions = {
+    theme: "vs",
+    language: "yaml",
+    renderLineHighlight: "none",
+    minimap: {
+        enabled: false,
+    },
+    scrollBeyondLastLine: false,
+};
+
+@Component({
+    selector: "app-yaml-editor",
+    templateUrl: "./yaml-editor.component.html",
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class YamlEditorComponent extends BaseEditorComponent implements OnChanges {
+    public readonly YAML_EDITOR_OPTIONS: monaco.editor.IStandaloneEditorConstructionOptions = YAML_EDITOR_OPTIONS;
+
+    @Output() public yamlTemplateChange = new EventEmitter<string>();
```

Comment:
There is a similar output in the base editor:

```
 @Output() public templateChange = new EventEmitter<string>();
```

Why not to reuse it and move the `modelChange` method to the base editor as well.

Reply:
Thank you, removed